File: VB/CustomControl.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
' ...

Namespace EUD_CustomProperty
	Friend Class DummyResFinder

	End Class

	<ToolboxItem(True)> _
	Public Class CustomControl
		Inherits XRControl

		Private customText_Renamed As String

		Public Sub New()
			customText_Renamed = String.Empty
		End Sub

		<Description("Gets or sets custom text."), DXDisplayName(GetType(DummyResFinder), "Resource1", "CustomControl.CustomText", "CustomText")> _
		Public Property CustomText() As String
			Get
				Return customText_Renamed
			End Get
			Set(ByVal value As String)
				customText_Renamed = value
			End Set
		End Property
	End Class

End Namespace

File: CS/CustomControl.cs
using System.ComponentModel;
using DevExpress.XtraReports.UI;
// ...

namespace EUD_CustomProperty {
    class DummyResFinder {

    }

    [ToolboxItem(true)]
    public class CustomControl : XRControl {

        private string customText;

        public CustomControl() {
            customText = string.Empty;
        }

        [
        Description("Gets or sets custom text."),
        DXDisplayName(typeof(DummyResFinder), "Resource1", "CustomControl.CustomText", "CustomText")
        ]
        public string CustomText {
            get { return customText; }
            set { customText = value; }
        }
    }

}
